components/helpers/ajax.js

```diff
@@ -0,0 +1,108 @@
+                if (Object.prototype.hasOwnProperty.call(settings.headers, header)) {
+                    request.setRequestHeader(header, settings.headers[header]);
+                }
+            });
+        }
+
+        request.onreadystatechange = function() {
+            if (request.readyState === 4) {
+                const reqText = (request.responseText || '');
+                const isJson = reqText.isJson();
+
+                if (request.status >= 200 && request.status < 300) {
+                    if (request.responseType === 'json') {
+                        resolve(request.response);
+                    } else if (settings.dataType === 'json' && isJson) {
+                        resolve(JSON.parse(reqText));
+                    } else {
+                        resolve(reqText);
+                    }
+                } else {
+                    if (request.status === 0 && url.startsWith('data:')) {
+                        // The request failed, probably due to unhelpful CORS policy in Safari.
+                        // See related WebKit bug https://bugs.webkit.org/show_bug.cgi?id=123978
+                        // Manually process the URL instead...
+                        console.log('Falling back to manual data: URI processing...');
+                        const parsedDataUri = parseDataUri(url);
+                        if (parsedDataUri) {
+                            resolve(settings.dataType === 'json' && parsedDataUri.isJson() ? JSON.parse(parsedDataUri) : parsedDataUri);
+                            return;
+                        }
+                    }
+                    try {
+                        reject([JSON.parse(reqText), request]);
+                    } catch (e) {
+                        reject([reqText, request]);
+                    }
+                }
+            }
+        };
+
+        // TODO: Rework this function, its gettting hard to follow!
+        if (type.toUpperCase() !== 'GET' && contentType === 'application/x-www-form-urlencoded' && typeof settings.data === 'object' && !isFormDataPost) {
+            data = objectToFormPost(data);
+        } else if (typeof settings.data !== 'string' && !isFormDataPost) { // dont stringify the browsers 'FormData' object
+            // stringify objects
+            data = JSON.stringify(settings.data);
+        }
+
+        try {
+            request.send(data || '');
+        } catch (e) {
+            console.log(e);
+        }
+    });
+
+    /**
+     * Manually extract the data from the URI.
+     * @param {string} href to target
+     * @returns {string} encoded data
+     */
+    function parseDataUri(href) {
+        const commaPos = href.indexOf(',');
+        // header is of the form data:[<mediatype>][;base64]
+        const header = href.substring(0, commaPos);
+        let data = href.substring(commaPos + 1);
+        if (header.endsWith(';base64')) {
+            data = escape(window.atob(data));
```

Comment:
Okay, so I add it as a bower dependency and reflected in the examples - https://github.com/cambridgeweblab/common-ui/pull/5/commits/ccb3bbd47d7f267135d68bafec68008b674acc43

I will resolve this amd issue as part of big-red-button since I already made ajax a dependency rather than a global (but I need this merge request first to avoid conflicts and duplicate work)